zerver/lib/users.py

```diff
@@ -65,6 +67,26 @@ def check_full_name(full_name_raw: str) -> str:
     # ban them.
     if re.search(r"\|\d+$", full_name_raw):
         raise JsonableError(_("Invalid format!"))
+
+    if require_unique_names(realm):
+        normalized_user_full_name = unicodedata.normalize("NFKC", full_name).lower()
```

Comment:
This should use [`.casefold()`](https://docs.python.org/3/library/stdtypes.html#str.casefold) rather than `.lower()`.